pkg/core/archive_experiment.go

```diff
@@ -14,13 +14,56 @@
 package core
 
 import (
+	"context"
+	"time"
+
 	"github.com/jinzhu/gorm"
+
+	"github.com/pingcap/chaos-mesh/api/v1alpha1"
 )
 
 // ExperimentStore defines operations for working with archive experiments
-type ExperimentStore interface{}
+type ExperimentStore interface {
+	// List returns an archive experiment list from the datastore.
+	List(context.Context) ([]*ArchiveExperiment, error)
+
+	// ListByKind returns an archive experiment list by kind from the datastore.
+	ListByKind(context.Context, string) ([]*ArchiveExperiment, error)
+
+	// Find returns an archive experiment by ID.
+	Find(context.Context, int64) (*ArchiveExperiment, error)
+
+	// FindByName returns an archive experiment by the name and namespace of the experiment.
+	FindByName(context.Context, string, string) (*ArchiveExperiment, error)
+
+	// Create persists a new archive experiment to the datastore.
+	Create(context.Context, *ArchiveExperiment) error
 
-// ArchiveExperiment represents a experiment instance.
+	// Update persists an updated experiment to the datastore.
+	Update(context.Context, *ArchiveExperiment) error
+}
+
+// ArchiveExperiment represents an experiment instance.
 type ArchiveExperiment struct {
+	ArchiveExperimentMeta
+	Experiment string `gorm:"size:2048"`
```

Comment:
What is this field? Does it store the chaos yaml or some others?

Reply:
This field will store the whole experiment struct.